schemas/adaptive-card.json

```diff
@@ -51,31 +66,48 @@
 				},
 				{
 					"$ref": "#/definitions/Action.OpenUrl"
+				},
+				{
+					"$ref": "#/definitions/Action.ToggleVisibility"
 				}
 			]
 		},
+		"ActionStyle": {
+			"type": "string",
+			"description": "Controls the style of an Action, which influences how the action is displayed, spoken, etc.",
+			"enum": [
+				"default",
+				"positive",
+				"destructive"
+			],
+			"version": "1.2"
+		},
 		"Action.OpenUrl": {
 			"additionalProperties": true,
 			"description": "When invoked, show the given url either by launching it in an external web browser or showing in-situ with embedded web browser.",
 			"properties": {
 				"type": {
 					"type": "string",
-					"description": "Must be `\"Action.OpenUrl\"`."
+					"description": "Must be `\"Action.OpenUrl\"`.",
+					"enum": [
+						"Action.OpenUrl"
+					]
 				},
 				"title": {
 					"type": "string",
 					"description": "Label for button or link that represents this action."
 				},
 				"iconUrl": {
 					"type": "string",
-					"format": "uri",
```

Comment:
>"format": "uri", [](start = 5, length = 16)

Why did we get rid of the uri format restriction on images? (this applies multiple places). The uri format supports data:image/png;base64,Rj9fj03092 as verified with the jsonschemavalidator.net

Reply:
Since we support specifying a baseURI in host config, non-fully qualified URIs are valid but don't validate with this requirement. See for example [https://github.com/microsoft/AdaptiveCards/blob/master/samples/v1.0/Elements/Image.ImageBaseUrl.json](https://github.com/microsoft/AdaptiveCards/blob/master/samples/v1.0/Elements/Image.ImageBaseUrl.json) 

---
In reply to: [289948025](https://github.com/microsoft/AdaptiveCards/pull/2998#discussion_r289948025) [](ancestors = 289948025)

Reply:
Ah :)

---
In reply to: [289951839](https://github.com/microsoft/AdaptiveCards/pull/2998#discussion_r289951839) [](ancestors = 289951839,289948025)